portsyncd/linksync.cpp

```diff
@@ -91,16 +109,30 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
     /* VLAN member: A separate entry in VLAN_TABLE will be inserted */
     if (master)
     {
-        key = m_ifindexNameMap[master] + ":" + key;
+        string member_key = m_ifindexNameMap[master] + ":" + key;
 
-        if (nlmsg_type == RTM_DELLINK)
-            m_vlanTableProducer.del(key);
+        if (nlmsg_type == RTM_DELLINK) /* Will it happen? */
+            m_vlanTableProducer.del(member_key);
         else /* RTM_NEWLINK */
         {
             FieldValueTuple t("tagging_mode", "untagged");
             fvVector.push_back(t);
```

Comment:
You can move fvVector to local block, and remove .clear(). Later you can define another fvVector, not to reuse the old one.